ruby/ql/lib/codeql/ruby/ApiGraphs.qll

```diff
@@ -389,6 +387,17 @@ module API {
           useStep(lbl, node, ref)
         )
       )
+      or
+      // `pred` is a use of class A
+      // `succ` is a use of class B
+      // there exists a class declaration B < A
+      exists(ClassDeclaration c, DataFlow::Node a, DataFlow::Node b |
+        use(pred, a) and
+        use(succ, b) and
+        resolveConstant(b.asExpr().getExpr()) = resolveConstantWriteAccess(c) and
+        c.getSuperclassExpr() = a.asExpr().getExpr() and
+        lbl = Label::subclass()
```

Comment:
I wonder if this should include the name of the sub class.

Reply:
That sounds reasonable. I can't immediately see what benefit it would bring though. Does it make anything particular easier/harder?

Reply:
Let's just skip it for now.